sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/workflow/ReflectiveWorkflowRouter.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl.workflow
+
+import com.google.protobuf.any.{ Any => ScalaPbAny }
+import kalix.javasdk.JsonSupport
+import kalix.javasdk.impl.MessageCodec
+import kalix.javasdk.impl.workflow.WorkflowRouter
+import kalix.javasdk.workflow.CommandContext
+import kalix.javasdk.workflow.Workflow
+import kalix.springsdk.impl.CommandHandler
+import kalix.springsdk.impl.InvocationContext
+import kalix.springsdk.impl.SpringSdkMessageCodec
+
+import java.lang.reflect.ParameterizedType
+
+class ReflectiveWorkflowRouter[S, W <: Workflow[S]](
+    override protected val workflow: W,
+    commandHandlers: Map[String, CommandHandler],
+    messageCodec: SpringSdkMessageCodec)
```

Comment:
This `messageCodec` is not being used and we pass the codec. 

Maybe we don't need to pass it to `decodeToAny` because already in scope?

Reply:
it is used now